lib/src/settings.rs

```diff
@@ -153,6 +180,10 @@ impl UserSettings {
         &self.config
     }
 
+    pub fn git_settings(&self) -> &GitSettings {
+        &self.git_settings
+    }
+
```

Comment:
True, it is cheap. Do you mean replacing this with

```
    pub fn git_settings(&self) -> GitSettings {
        self.git_settings.clone()
    }
```

? That doesn't seem to make any of the current use cases easier (probably because they're all read-only).

Reply:
It doesn't simplify for the callers, but it simplifies by not having to store the `UserSettings::git_settings` field. It's not a big simplification, of course, but there doesn't seem to be much more reason to have that field than to have e.g. `color_config` or `graph_style` or ...

Reply:
Ah, you mean like parsing it from config anew. Yeah sounds good.